ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -185,6 +191,8 @@ private async Task<ImportResult> ValidateAndImportNotificationGroupAsync(Perform
                 await _migratedNotificationsMarker.MarkNotificationsAsImportedAsync(savedNotifications);
                 importResult.NtbsIds = savedNotifications.ToDictionary(x => x.LegacyId, x => x.NotificationId);
                 await ImportReferenceLabResultsAsync(context, requestId, savedNotifications, importResult);
+                await _cultureAndResistanceService.MigrateNotificationCultureResistanceSummary(savedNotifications);
```

Comment:
I'm a little unclear on how this migration used to happen... was it part of a stored procedure in the reporting or specimen matching database?

Reply:
Currently this part of the migration didn't happen as part of the migration per-say, instead these last bits of information were recalculated as part of the `reporting-data-refresh` and `drug-resistance-profile-update` jobs that night. 

In particular the bits that weren't migrated until the following night were:
* Culture and Resistance summaries, shown at the top of the Test Results sections as "Reference laboratory results". The specimens that are being summarised here are copied over by `NotificationImportService.ImportReferenceLabResultsAsync`. Before this would just get calculated in the next run of `uspGenerate` in the reporting DB, though this data is now being stored in specimen-matching. I've added an SP here to duplicate the summary for the legacy notification to have the newly-migrated NTBS notification ID, which is what the table would look like after `uspGenerate` is run but this is much quicker. The app queries directly from this specimen-matching table when displaying test results.
* Drug resistance profile, and species, which is part of the culture and resistance summary but also cached in the NTBS `DrugResistanceProfile` table. Before this would get copied over using the Hangfire job that night (after the summary had been calculated by the `reporting-data-refresh` job. Now the data is available right away due to my changes, so I trigger a `DrugResistanceProfile` update just on these newly migrated notifications to bring the profile in right away.

Does that all make sense? Sorry for not explaining it clearly before!

Reply:
Great thanks, it's good to have this extra context! As discussed, my worry was just making sure that the two sources of this data stay in sync, and (as you said) this is true because the summaries are only calculated in one place, we're just duplicating them here for NTBS notifications. So all good!